classes/image_types_ota.bbclass

```diff
@@ -63,13 +63,23 @@ IMAGE_CMD_ota () {
 
 	ostree_target_hash=$(cat ${WORKDIR}/ostree_manifest)
 
+	# Use OSTree hash to avoid any potential race conditions between
+	# multiple builds accessing the same ${OSTREE_REPO}.
 	ostree --repo=${OTA_SYSROOT}/ostree/repo pull-local --remote=${OSTREE_OSNAME} ${OSTREE_REPO} ${ostree_target_hash}
 	kargs_list=""
 	for arg in ${OSTREE_KERNEL_ARGS}; do
 		kargs_list="${kargs_list} --karg-append=$arg"
 	done
 
-	ostree admin --sysroot=${OTA_SYSROOT} deploy ${kargs_list} --os=${OSTREE_OSNAME} ${ostree_target_hash}
+	# Create the same reference on the device we use in the archive OSTree
+	# repo in ${OSTREE_REPO}. This reference will show up when showing the
+	# deployment on the device:
+	# ostree admin status
+	# If a remote with the name ${OSTREE_OSNAME} is configured, this also
+	# will allow to use:
+	# ostree admin upgrade
+	ostree --repo=${OTA_SYSROOT}/ostree/repo refs --create=${OSTREE_OSNAME}:${OSTREE_BRANCHNAME} ${ostree_target_hash}
+	ostree admin --sysroot=${OTA_SYSROOT} deploy ${kargs_list} --os=${OSTREE_OSNAME} ${OSTREE_OSNAME}:${OSTREE_BRANCHNAME}
```

Comment:
Cool! That's definitely better than #687

Reply:
@patrickvacek currently I add the remote manually for testing, but we might add a recipe which then adds a default remote for a pure OSTree use-case.